mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -75,6 +75,10 @@ static tosa::CastOp createCastOp(PatternRewriter &rewriter, Location loc,
   return op;
 }
 
+static Type getShapedElementTy(const Value &v) {
```

Comment:
`Value` is a smart pointer, there's no need to `const Value &` it.

Reply:
Good point!

> which definitions of quantize and de-quantize do frameworks actually want?

Turns out none of framework want this, it driven exclusively by migraphx integration. The prototype used to sent to us is wrong.

Reply:
... ok, so what _do_ the frameworks want? And why does MIGraphX want this weird thing?

(I'm willing to approve on the basis that this is important integration stuff, but I'd like to understand)

Reply:
> so what do the frameworks want?

int8 optimizations shoot for inference, framework doesn't want it because they care about training instead.

> And why does MIGraphX want this weird thing?

Not sure how is it "weird"? I'd agree that they can implement this in a more clever way, like TOSA rescale, but the math they have looks reasonable to me. It is one way to do it, but why they do what they do is out of scope for this review.